src/processing.rs

```diff
@@ -193,16 +202,14 @@ This function the derivative features.
 */
 pub fn derivative_extraction(feat: Array2<f32>, DeltaWindows: i32) -> Array2<f32> {
     // Getting the shape of the vector.
-    let (rows, cols) = feat.shape;
+    let (rows, cols) = feat.shape();
 
     // Difining the vector of differences.
-    let mut DIF = np.zeros(feat.shape, dtype = feat.dtype);
+    let mut DIF = ndarray::ArrayBase::<f32>::zeros(feat.shape());
     let Scale = 0;
 
     // Pad only along features in the vector.
-    let FEAT = np
-        .lib
-        .pad(feat, ((0, 0), (DeltaWindows, DeltaWindows)), "edge");
+    let FEAT = ndarray_ndimage::pad(feat, ((0, 0), (DeltaWindows, DeltaWindows)), "edge");
```

Comment:
ndarray_ndimage::pad looks for a three dimensional array here, AFAIK we are working with one dimension and two dimensions. They offer a solution in https://github.com/rust-ndarray/ndarray/issues/823, though I'd say you've already seen that link.